Shared/Models/Message Model/MessageKind.swift

```diff
@@ -94,14 +88,42 @@ extension MessageKind: Equatable {
             return lhsAudio == rhsAudio
         case (.contact(let lhsContact), .contact(let rhsContact)):
             return lhsContact == rhsContact
-        case (.link(let lhsLink), .link(let rhsLink)):
-            return lhsLink == rhsLink
+        case (.link(let lhsURL, let lhsStringURL), .link(let rhsURL, let rhsStringURL)):
+            return lhsURL == rhsURL && lhsStringURL == rhsStringURL
```

Comment:
Distinguishing between the message text, and the cleaned up/sanitized url object.